torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1126,6 +1115,17 @@ c10::intrusive_ptr<ProcessGroup::Work> ProcessGroupNCCL::collective(
   work->opTimeout_ = opTimeout_;
   work->store_ = store_;
 
+  if (work->recordFunctionEndCallback_) {
```

Comment:
Thanks for working on this @rohan-varma . Would you elaborate why it matters to add callback later here?

Reply:
@mrzzd As discussed offline, it's equivalent to directly invoking the callback inline since the callback doesn't do any GPU work. See https://github.com/pytorch/pytorch/pull/48561 for more of a discussion/context.